Classes/View/PreviewView.php

```diff
@@ -671,13 +671,32 @@ protected function parseGridColumnTemplate(array $row, Column $column, $colPosFl
 				$label = $column->getLabel();
 			}
 		}
+
+		// use this contents language as default
+		$columnLanguageUid = $row['sys_language_uid'];
+
+		// but for language mode all (uid -1):
+		if ( $columnLanguageUid == -1 ) {
+			/** @var \TYPO3\CMS\Backend\Controller\PageLayoutController $pageLayoutController */
+			$pageLayoutController = $GLOBALS['SOBE'];
+			$isColumnView    = ( 1 == $pageLayoutController->MOD_SETTINGS['function'] );
+			$isLanguagesView = ( 2 == $pageLayoutController->MOD_SETTINGS['function'] );
+			if ($isColumnView) {
+				$columnLanguageUid = $pageLayoutController->current_sys_language;
+			} elseif ($isLanguagesView) {
+				// if a language-all (uid -1) grid element in lanugages view
```

Comment:
This comment should probably be improved a bit, the descriptions of conditions are unclear (at least to me). It's better to describe here *why* we have to use the default language UID.